metagpt/roles/role.py

```diff
@@ -478,10 +478,10 @@ async def _react(self) -> Message:
 
     async def _plan_and_act(self) -> Message:
         """first plan, then execute an action sequence, i.e. _think (of a plan) -> _act -> _act -> ... Use llm to come up with the plan dynamically."""
-
-        # create initial plan and update it until confirmation
-        goal = self.rc.memory.get()[-1].content  # retreive latest user requirement
-        await self.planner.update_plan(goal=goal)
+        if not self.planner.plan.goal:
```

Comment:
what's the case for this condition check

Reply:
when recovered, not need to update plan if goal already existed

Reply:
otherwise it will replan